client/js/widgets/widget.js

```diff
@@ -1720,6 +1720,10 @@ export class Widget extends StateManaged {
             return w.get(a.property) === a.value;
           });
 
+          if (a.random) {
+            c = shuffleArray(Object.values(c));
```

Comment:
Does the `Object.values` serve a purpose?

Reply:
It didn't work for me otherwise.  The function is looking for an array and c is an object and this was my effort to turn it into an array.  I'm sure I missed something obvious.

Reply:
Hmm, it really shouldn't be an object. Apart from the fact that every array is also an object.

I can have a look tomorrow..

Reply:
Of course it works now.  I must not have set up my test room properly.  Fixed.